README.md

```diff
@@ -26,30 +23,130 @@ dune-release help release # for help about releasing your package
 dune-release help         # for more help
 ```
 
-[doc]: https://samoht.github.com/dune-release/
+Calling `dune-release` without any argument will start the automated release process, composed of the following steps:
+- create the distribution archive;
+- publish it online with its documentation;
+- create an opam package;
+- submit it to OCaml's opam repository.
+
+The most basic workflow is:
+
+```
+dune-release tag
+dune-release
+```
+
+Each step is refined and explained with more details below.
+If this is your first time using `dune-release` you might choose to run the `dune-release` commands with the `--dry-run` argument, so that no action is actually performed but it allows you to check what would be done. And then run the `dune-release` commands without `--dry-run` when you are sure of what you are doing.
+
+
+### Tag the distribution with a version
+
+This step should be executed prior to running the `dune-release` automated process.
+
+The tagging command of `dune-release` will extract the latest version tag from the package's change log and tag the VCS HEAD commit with it if it is invoked without argument:
+
+```
+dune-release tag
+```
+
+This will only work if the change log follows a certain format. If you do not want to rely on this extraction you can specify it on the command line:
```

Comment:
Might be worth adding a section about that to explain what format dune-release expects and what it does with it (ie infer the tag/version and generate a publication message).

Reply:
done